pkg/kubernetes/controller.go

```diff
@@ -2,11 +2,66 @@ package kubernetes
 
 import (
 	"context"
+	"errors"
 
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	networkv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 )
 
 func (k *Kubernetes) GetPodStatus(namespace, name string) (string, error) {
-	pod, err := k.client.CoreV1().Pods(namespace).Get(context.TODO(), name, v1.GetOptions{})
+	pod, err := k.client.CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 	return string(pod.Status.Phase), err
 }
+
+func (k *Kubernetes) CreateStatefulSet(namespace string, object runtime.Object) (*appsv1.StatefulSet, error) {
```

Comment:
In my opinion, we should not use abstract type as the input params for Create functions, handle the type of object before calling helps us keep the code cleaner. ^^

Reply:
fix it! thank you